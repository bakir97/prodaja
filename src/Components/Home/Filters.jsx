import React from "react";
import { TextField, InputAdornment, Grid } from "@material-ui/core";
const style = {
  start: { paddingLeft: 5, paddingTop: 10, textAlign: "start" },
  end: { paddingRight: 5, paddingTop: 10, textAlign: "end" },
  root: { padding: 10 }
};
export default class App extends React.Component {
  state = {
    minCijena: this.props.minCijena,
    maxCijena: this.props.maxCijena,
    minTezina: this.props.minTezina,
    maxTezina: this.props.maxTezina
  };
  changeCijenaMin = e => {
    this.props.cijenaFilters({ minCijena: e.target.value });
  };
  changeCijenaMax = e => {
    this.props.cijenaFilters({ maxCijena: e.target.value });
  };
  changeTezinaMin = e => {
    this.props.cijenaFilters({ minTezina: e.target.value });
  };
  changeTezinaMax = e => {
    this.props.cijenaFilters({ maxTezina: e.target.value });
  };
  render() {
    const { minCijena, maxCijena, minTezina, maxTezina } = this.state;
    return (
      <Grid
        container
        justify="center"
        alignContent="center"
        alignItems="center"
        style={style.root}
      >
        <Grid item xs={6} style={style.end}>
          <TextField
            id="minCijena"
            type="number"
            variant="outlined"
            label="minCijena"
            value={minCijena}
            onChange={e => this.changeCijenaMin(e)}
            InputProps={{
              endAdornment: <InputAdornment position="end">KM</InputAdornment>
            }}
          />
        </Grid>

        <Grid item xs={6} style={style.start}>
          <TextField
            id="maxCijena"
            type="number"
            variant="outlined"
            label="maxCijena"
            value={maxCijena}
            onChange={e => this.changeCijenaMax(e)}
            InputProps={{
              endAdornment: <InputAdornment position="end">KM</InputAdornment>
            }}
          />
        </Grid>
        <Grid item xs={6} style={style.end}>
          <TextField
            id="minTezina"
            type="number"
            variant="outlined"
            label="minTezina"
            value={minTezina}
            onChange={e => this.changeTezinaMin(e)}
            InputProps={{
              endAdornment: <InputAdornment position="end">KG</InputAdornment>
            }}
          />
        </Grid>
        <Grid item xs={6} style={style.start}>
          <TextField
            id="maxTezina"
            type="number"
            variant="outlined"
            label="maxTezina"
            value={maxTezina}
            onChange={e => this.changeTezinaMax(e)}
            InputProps={{
              endAdornment: <InputAdornment position="end">KG</InputAdornment>
            }}
          />
        </Grid>
      </Grid>
    );
  }
}
