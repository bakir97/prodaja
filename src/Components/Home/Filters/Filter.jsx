import React from "react";
import { TextField, InputAdornment, Grid } from "@material-ui/core";
const style = {
  start: { paddingLeft: 5, paddingTop: 10, textAlign: "start" },
  end: { paddingRight: 5, paddingTop: 10, textAlign: "end" }
};
const sameAtt = {
  type: "number",
  variant: "outlined",
  InputProps: {
    endAdornment: <InputAdornment position="end">KM</InputAdornment>
  }
};
export default ({ position, ...other }) => {
  return (
    <Grid item xs={6} style={position === "start" ? style.start : style.end}>
      <TextField {...other} {...sameAtt} />
    </Grid>
  );
};
