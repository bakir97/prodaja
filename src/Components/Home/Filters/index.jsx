import React from "react";
import { Grid } from "@material-ui/core";
import Filters from "./Filters";
const style = {
  root: { padding: 10 }
};
export default class App extends React.Component {
  state = {
    minCijena: this.props.minCijena,
    maxCijena: this.props.maxCijena,
    minTezina: this.props.minTezina,
    maxTezina: this.props.maxTezina
  };
  changeCijenaMin = e => {
    this.props.filters({ minCijena: e.target.value });
  };
  changeCijenaMax = e => {
    this.props.filters({ maxCijena: e.target.value });
  };
  changeTezinaMin = e => {
    this.props.filters({ minTezina: e.target.value });
  };
  changeTezinaMax = e => {
    this.props.filters({ maxTezina: e.target.value });
  };
  render() {
    const { minCijena, maxCijena, minTezina, maxTezina } = this.state;
    return (
      <Grid
        container
        justify="center"
        alignContent="center"
        alignItems="center"
        style={style.root}
      >
        <Filters
          minCijena={minCijena}
          maxCijena={maxCijena}
          minTezina={minTezina}
          maxTezina={maxTezina}
          changeCijenaMin={this.changeCijenaMin}
          changeCijenaMax={this.changeCijenaMax}
          changeTezinaMin={this.changeTezinaMin}
          changeTezinaMax={this.changeTezinaMax}
        />
      </Grid>
    );
  }
}
