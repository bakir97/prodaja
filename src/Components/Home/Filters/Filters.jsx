import React from "react";
import Filter from "./Filter";
const Filters = ({
  minCijena,
  maxCijena,
  minTezina,
  maxTezina,
  changeCijenaMin,
  changeCijenaMax,
  changeTezinaMin,
  changeTezinaMax
}) => {
  return (
    <>
      <Filter
        position="end"
        id="minCijena"
        label="minCijena"
        value={minCijena}
        onChange={e => changeCijenaMin(e)}
      />
      <Filter
        position="start"
        id="maxCijena"
        label="maxCijena"
        value={maxCijena}
        onChange={e => changeCijenaMax(e)}
      />
      <Filter
        position="end"
        id="minTezina"
        label="minTezina"
        value={minTezina}
        onChange={e => changeTezinaMin(e)}
      />
      <Filter
        position="start"
        id="maxTezina"
        label="maxTezina"
        value={maxTezina}
        onChange={e => changeTezinaMax(e)}
      />
    </>
  );
};

export default Filters;
