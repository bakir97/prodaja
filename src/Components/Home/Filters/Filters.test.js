import Filters from "./Filters";
describe("Filters", () => {
  const wrapper = shallow(<Filters />);
  it("should render", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
