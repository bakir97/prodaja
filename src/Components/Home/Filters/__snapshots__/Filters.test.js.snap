// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should render 1`] = `
<Fragment>
  <_default
    id="minCijena"
    label="minCijena"
    onChange={[Function]}
    position="end"
  />
  <_default
    id="maxCijena"
    label="maxCijena"
    onChange={[Function]}
    position="start"
  />
  <_default
    id="minTezina"
    label="minTezina"
    onChange={[Function]}
    position="end"
  />
  <_default
    id="maxTezina"
    label="maxTezina"
    onChange={[Function]}
    position="start"
  />
</Fragment>
`;
